core/src/epoch_slots.rs

```diff
@@ -84,6 +84,9 @@ impl Uncompressed {
             (min_slot - self.first_slot) as usize
         };
         for i in start..self.num {
+            if i >= self.slots.len() as usize {
```

Comment:
seems like this: https://github.com/solana-labs/solana/blob/master/core/src/epoch_slots.rs#L105, should guard this call, are we missing an edge case somewhere?

Reply:
That probably doesn't get called if we deserialize the object from the network right?

Reply:
@sakridge yeah, somebody could give a bad length, but was that the case here? (was it malicious or just an edge case we're missing)?

Reply:
Also side note, should this be https://github.com/solana-labs/solana/blob/master/core/src/epoch_slots.rs#L101

`self.slots.len()`, not `self.slots.capacity()`?

Reply:
I think capacity is right.  Otherwise it should set the bit if it can.

Reply:
https://github.com/solana-labs/solana/blob/master/core/src/epoch_slots.rs#L76, hmm but just like a `Vec`,  `len -1 ` is the maximal element you can access without an out of range exception. The `capacity` can be greater than the len(), but doesn't determine the range?

https://docs.rs/bv/0.10.0/bv/struct.BitVec.html#method.capacity
https://docs.rs/bv/0.10.0/bv/struct.BitVec.html#method.len

Reply:
@carllin len is correct in this case, because even if capacity is higher none of those values are 1.

Reply:
> @sakridge yeah, somebody could give a bad length, but was that the case here? (was it malicious or just an edge case we're missing)?

Malicious

Reply:
@aeyakovenko yea, but calling `set` on an index `>= len` should panic with out of bounds. It's interesting...so it works right now because `len == capacity` always when you call `new_fill` with a size that is a multiple of 8, which is what we're doing in `Uncompressed::new()`